dlt/destinations/impl/clickhouse/clickhouse.py

```diff
@@ -0,0 +1,277 @@
+
+        if not bucket_path:
+            # Local filesystem.
+            raise NotImplementedError("Only object storage is supported.")
+
+        bucket_url = urlparse(bucket_path)
+        bucket_scheme = bucket_url.scheme
+
+        table_function: str
+
+        if bucket_scheme in ("s3", "gs", "gcs"):
+            bucket_http_url = convert_storage_to_http_scheme(bucket_url)
+
+            table_function = (
+                render_s3_table_function(
+                    bucket_http_url,
+                    staging_credentials.aws_secret_access_key,
+                    staging_credentials.aws_secret_access_key,
+                    file_format=file_extension,  # type: ignore[arg-type]
+                )
+                if isinstance(staging_credentials, AwsCredentialsWithoutDefaults)
+                else render_s3_table_function(
+                    bucket_http_url,
+                    file_format=file_extension,  # type: ignore[arg-type]
+                )
+            )
+        elif bucket_scheme in ("az", "abfs"):
+            if isinstance(staging_credentials, AzureCredentialsWithoutDefaults):
+                # Authenticated access.
+                account_name = staging_credentials.azure_storage_account_name
+                storage_account_url = (
+                    f"{staging_credentials.azure_storage_account_name}.blob.core.windows.net"
+                )
+                account_key = staging_credentials.azure_storage_sas_token
+                container_name = bucket_url.netloc
+                blobpath = bucket_url.path
+
+                format_mapping = {"jsonl": "JSONEachRow", "parquet": "Parquet"}
+                clickhouse_format = format_mapping[file_extension]
+
+                table_function = (
+                    f"azureBlobStorage('{storage_account_url}','{container_name}','{ blobpath }','{ account_name }','{ account_key }','{ clickhouse_format}')"
+                )
+
+            else:
+                # Unsigned access.
+                raise NotImplementedError(
+                    "Unsigned Azure Blob Storage access from Clickhouse isn't supported as yet."
+                )
+
+        with client.begin_transaction():
+            client.execute_sql(
+                f"""INSERT INTO {qualified_table_name} SELECT * FROM {table_function}"""
+            )
+
+    def state(self) -> TLoadJobState:
+        return "completed"
+
+    def exception(self) -> str:
+        raise NotImplementedError()
+
+
+class ClickhouseClient(SqlJobClientWithStaging, SupportsStagingDestination):
+    capabilities: ClassVar[DestinationCapabilitiesContext] = capabilities()
+
+    def __init__(
+        self,
+        schema: Schema,
+        config: ClickhouseClientConfiguration,
+    ) -> None:
+        # TODO: There are no schemas in Clickhouse. No point in having schemas, only dataset names and table names for example "dataset1_mytable".
```

Comment:
I'm not sure what the todo is here, but we have the schema anyway and I don't think it is harmful to have it a as a property on the client. At least this is my take.

Reply:
You are 100% right. I've opted for using underscore and the separator: `database`.`{schema/dataset}_tablename`. 

https://github.com/dlt-hub/dlt/blob/86685fd684bff29c8c016d9f20bfb4b5d9055e8d/dlt/destinations/impl/clickhouse/sql_client.py#L154

Thoughts? Is this robust enough for a first draft?